compiler/include/expr.h

```diff
@@ -474,7 +474,10 @@ static inline bool needsCapture(FnSymbol* taskFn) {
 }
 
 inline Symbol* ShadowVarSymbol::outerVarSym() const {
-  return this->outerVarSE()->symbol();
+  if (SymExpr* ovse = this->outerVarSE())
```

Comment:
It's not clear to me why you needed this change, but at least there's no reason for the assignment within the conditional. e.g.
```
SymExpr* ovse = this->outerVarSE();
if (ovse != NULL) return ovse->symbol();
else return NULL;
```

But it's obviously a style thing.

Reply:
This change is necessary because this->outerVarSE() can be NULL.

My reason for having a declaration within a conditional is that this style is quite common in our compiler. Now that I am used to it, I find it quite elegant.